File: VoxaScriptJB/src/main/java/homeworkstudios/language/VoxaScriptFormattingModelBuilder.java
package homeworkstudios.language;

import com.intellij.formatting.FormattingContext;
import com.intellij.formatting.FormattingModel;
import com.intellij.formatting.FormattingModelBuilder;
import com.intellij.formatting.FormattingModelProvider;
import org.jetbrains.annotations.NotNull;

public class VoxaScriptFormattingModelBuilder implements FormattingModelBuilder {
    @Override
    public @NotNull FormattingModel createModel(FormattingContext formattingContext) {
        return FormattingModelProvider.createFormattingModelForPsiFile(
                formattingContext.getContainingFile(),
                new VoxaScriptBlock(formattingContext.getNode(), null, null),
                formattingContext.getCodeStyleSettings()
        );
    }
}

File: VoxaScriptJB/src/main/java/homeworkstudios/language/VoxaScriptFormattingModelBuilder.kt
import com.intellij.formatting.FormattingContext
import com.intellij.formatting.FormattingModel
import com.intellij.formatting.FormattingModelBuilder
import com.intellij.formatting.FormattingModelProvider
import homeworkstudios.language.VoxaScriptBlock

class VoxaScriptFormattingModelBuilder : FormattingModelBuilder {
    override fun createModel(formattingContext: FormattingContext): FormattingModel {
        return FormattingModelProvider.createFormattingModelForPsiFile(
                formattingContext.containingFile,
                VoxaScriptBlock(formattingContext.node, null, null),
                formattingContext.codeStyleSettings
        )
    }
}